crates/polars-ops/src/chunked_array/strings/namespace.rs

```diff
@@ -357,6 +358,39 @@ pub trait Utf8NameSpaceImpl: AsUtf8 {
         Ok(out)
     }
 
+    /// Count all successive non-overlapping regex matches.
+    fn count_match_many(&self, pat: &Utf8Chunked) -> PolarsResult<UInt32Chunked> {
+        let ca = self.as_utf8();
+        polars_ensure!(
+            ca.len() == pat.len(),
+            ComputeError: "pattern's length: {} does not match that of the argument series: {}",
+            pat.len(), ca.len(),
+        );
+
+        // Very simple cache: don't recompile the same regex for multiple values in a row.
+        // TODO: proper LRU cache with reasonable memory limit.
+        let mut reg = Regex::new("").unwrap();
+        let mut lastpat = "";
+
+        let out: UInt32Chunked = ca
+            .into_iter()
+            .zip(pat)
+            .map(|(opt_s, opt_pat)| match (opt_s, opt_pat) {
+                (Some(s), Some(pat)) => {
+                    if pat != lastpat {
+                        reg = Regex::new(pat)?;
+                        lastpat = pat;
+                    }
+                    Ok(Some(reg.find_iter(s).count() as u32))
+                },
+                _ => Ok(None),
+            })
+            // Return early on error
```

Comment:
I don't think this comment is necessary, otherwise we'd be adding this... almost everywhere we do a collect. Also, please double-check the other comments to make sure they proper sentences (they start with a capital letter and end in a period).